fronts-client/src/util/shared.ts

```diff
@@ -77,14 +77,28 @@ const addGroupsForStage = (
 		groupsWithNames.push(createGroup(null, null, getAllCards(groups)));
 	}
 
-	// Finally we need to sort the groups according to their ids.
+	// We need to sort the groups according to their ids.
 	const sortedGroupsWithNames = sortBy(
 		groupsWithNames,
 		(group) => -getGroupIndex(group.id),
 	);
+
+	// Finally, we need to filter out any groups that have maxItems set to 0 (e.g. in flex gen).
+	const sortedNamedGroupsWithoutMaxItemSetToZero =
+		!collectionConfig.groupsConfig
+			? sortedGroupsWithNames
+			: sortedGroupsWithNames.filter((group) => {
```

Comment:
Personal preference so feel free to ignore: 
I find early returns easier to parse eg
```
if (!collectionConfig.groupsConfig) return sortedGroupsWithNames;
return sortedGroupsWithNames.filter(...........
```